src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -153,7 +153,7 @@ public Flux<ModificationInfos> applyGroovyScript(UUID networkUuid, String varian
             getNetworkModificationInfos(networkUuid, variantId).flatMapIterable(networkInfos -> {
                 NetworkStoreListener listener = NetworkStoreListener.create(networkInfos.getNetwork(), networkUuid, groupUuid, networkModificationRepository, equipmentInfosService, false, networkInfos.isApplyModifications());
 
-                return execApplyGroovyScript(listener, groovyScript, reportUuid);
+                return execApplyGroovyScript(listener, groovyScript, reportUuid, groupUuid);
             })
         );
     }
```

Comment:
You forgot to set groupUuid as taskKey for the reporter model in execChangeSwitchState ...

Reply:
Done